exercises/queen-attack/example.rs

```diff
@@ -0,0 +1,48 @@
+#[derive(Debug, PartialEq)]
+pub struct Queen {
+    position: (i8, i8)
+}
+
+#[derive(Debug, PartialEq)]
+pub struct InvalidQueen;
+
+impl Queen {
+    pub fn new(position: (i8, i8)) -> Result<Queen, InvalidQueen> {
+        if (position.0 < 0 || position.0 > 7) || (position.1 < 0 || position.1 >7 ) {
+            Err(InvalidQueen)
+        } else {
+            Ok(Queen {position: position})
+        }
+    }
+}
+
+pub struct Queens {
+    first: Queen,
+    second: Queen
+}
+
+impl Queens {
+    pub fn new(first: Result<Queen, InvalidQueen>, second: Result<Queen, InvalidQueen>) -> Queens {
```

Comment:
I'm totally open to changing the interface. If I were writing this from scratch in Ruby I probably would do `queen1.can_attack?(queen2)`.

I think my API was mostly being led by the other implementations of this exercise, which tend to look like 

```ruby
queens = Queens.new(white: [2, 3], black: [4, 7])
assert !queens.attack?
```

But since I'm trying to re-think which tests are appropriate, I might as well also re-think what API is appropriate. Let me take a shot at the `queen1.can_attack(queen2)` approach.